### PR TITLE
[HOPSWORKS-546] Multiple Jupyter notebooks opened for the same user cause PersistenceException

### PR DESCRIPTION
**Pull request by robzor92**

### CODE REVIEW

**Comment by robzor92** [comment]
https://github.com/logicalclocks/hops-testing/pull/276

File: files/default/sql/ddl/0.9.0__initial_tables.sql [hunk 1]
```diff
@@ -617,6 +617,7 @@ CREATE TABLE `jupyter_project` (
   `pid` bigint(20) NOT NULL,
   `project_id` int(11) NOT NULL,
   PRIMARY KEY (`port`),
+  UNIQUE KEY `unique_hdfs_user` (`hdfs_user_id`),
   KEY `hdfs_user_idx` (`hdfs_user_id`),
```
Comment (SirOibaf) [inline files/default/sql/ddl/0.9.0__initial_tables.sql]: We should condense the `unique key` and the `key` in the `hdfs_user_id`. There is no point having both

File: files/default/sql/ddl/updates/0.9.0.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+ALTER TABLE jupyter_project ADD CONSTRAINT unique_hdfs_user UNIQUE (hdfs_user_id);
```
Comment (SirOibaf) [inline files/default/sql/ddl/updates/0.9.0.sql]: Same as above

File: files/default/sql/ddl/0.9.0__initial_tables.sql (hunk 1 above)
Comment (robzor92) [reply files/default/sql/ddl/0.9.0__initial_tables.sql]: from what I understand key and unique key are different. key creates a DB index while unique key is a constraint.

Comment (robzor92) [reply files/default/sql/ddl/0.9.0__initial_tables.sql]: fixed

File: files/default/sql/ddl/updates/0.9.0.sql (hunk 2 above)
Comment (robzor92) [reply files/default/sql/ddl/updates/0.9.0.sql]: fixed

File: files/default/sql/ddl/0.9.0__initial_tables.sql (hunk 1 above)
Comment (SirOibaf) [reply files/default/sql/ddl/0.9.0__initial_tables.sql]: You cannot assume the name of the constraint. You have to query the information schema to get the correct name by querying the `information_schema` db

File: files/default/sql/ddl/updates/0.9.0.sql [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+
+ALTER TABLE jupyter_project DROP FOREIGN KEY `FK_103_525`;
```
Comment (SirOibaf) [inline files/default/sql/ddl/updates/0.9.0.sql]: You cannot assume the name of the constraint. You have to query the information schema to get the correct name by querying the `information_schema` db

Comment (robzor92) [reply files/default/sql/ddl/updates/0.9.0.sql]: Added the sql for querying, migration and rollback both work

File: files/default/sql/ddl/updates/0.9.0.sql [hunk 4]
```diff
@@ -0,0 +1,13 @@
+ALTER TABLE jupyter_project ADD CONSTRAINT unique_hdfs_user UNIQUE (hdfs_user_id);
+
+-- Find the name of the FK
+SET @fk_name = (SELECT CONSTRAINT_NAME FROM INFORMATION_SCHEMA.KEY_COLUMN_USAGE WHERE TABLE_SCHEMA = "hopsworks" AND TABLE_NAME = "jupyter_project" AND REFERENCED_TABLE_NAME="hdfs_users");
```
Comment (tkakantousis) [inline files/default/sql/ddl/updates/0.9.0.sql]: @robzor92 Just fyi, that can return multiple constraints. In the future might be safer to add referenced column name as well. We can merge it.

**Pull request merged**